sinks/gcm/sink.go

```diff
@@ -31,24 +37,233 @@ func NewSink() (*GcmSink, error) {
+	labels[labelHostname] = pod.Hostname
+
+	// Break the individual metrics from the container statistics.
+	for _, container := range pod.Containers {
+		metrics = append(metrics, self.containerToMetrics(container, labels)...)
+	}
+	return metrics
+}
+
+func (self *GcmSink) rawContainersToMetrics(containers []sources.RawContainer) []Metric {
+	metrics := make([]Metric, 0, len(containers))
 
-	return nil
+	labels := make(map[string]string)
+	for _, container := range containers {
+		labels[labelHostname] = container.Hostname
+		metrics = append(metrics, self.containerToMetrics(&container.Container, labels)...)
+	}
+	return metrics
+}
+
+func (self *GcmSink) containerToMetrics(container *sources.Container, labels map[string]string) []Metric {
+	labels[labelContainerName] = container.Name
+
+	// One metric value per data point.
+	metrics := make([]Metric, 0, len(container.Stats))
+	for _, stat := range container.Stats {
+		// Add all supported metrics that have values.
+		for _, supported := range self.supportedMetrics {
+			if supported.HasValue(&container.Spec) {
+				// Cumulative stats have container creation time as start time.
+				var startTime time.Time
+				if supported.Type == MetricCumulative {
+					startTime = container.Spec.CreationTime
+				} else {
+					startTime = stat.Timestamp
+				}
+
+				metrics = append(metrics, Metric{
+					Name:   supported.Name,
+					Labels: labels,
+					Start:  startTime,
+					End:    stat.Timestamp,
+					Value:  supported.GetValue(stat),
+				})
+			}
+		}
+	}
+	return metrics
 }
 
 func (self *GcmSink) GetConfig() string {
 	desc := "Sink type: GCM\n"
+
+	// Add metrics being exported.
+	desc += "\tExported metrics:"
+	for _, supported := range self.supportedMetrics {
+		desc += fmt.Sprintf("\t\t%s: %s", supported.Name, supported.Description)
+	}
+
+	// Add labels being used.
+	desc += "\tExported labels:"
+	for _, label := range allLabels {
+		desc += fmt.Sprintf("\t\t%s: %s", label.Key, label.Description)
+	}
+
 	desc += "\n"
 	return desc
 }
+
+const (
+	labelPodId         = "pod_id"
+	labelContainerName = "container_name"
+	labelLabels        = "labels"
+	labelHostname      = "hostname"
+)
+
+// TODO(vmarmol): Things we should consider adding (note that we only get 10 labels):
+// - POD name, container name, and host IP: Useful to users but maybe we should just mangle them with ID and IP
+// - Namespace: Are IDs unique only per namespace? If so, mangle it into the ID.
+var allLabels = []LabelDescriptor{
+	LabelDescriptor{
+		Key:         labelPodId,
+		Description: "The unique ID of the pod",
+	},
+	LabelDescriptor{
+		Key:         labelContainerName,
+		Description: "User-provided name of the container or full container name for system containers",
+	},
+	LabelDescriptor{
+		Key:         labelLabels,
+		Description: "Comma-separated list of user-provided labels",
+	},
+	LabelDescriptor{
+		Key:         labelHostname,
+		Description: "Hostname where the container ran",
```

Comment:
To be unambiguous this would be the "hostname of the node where the container ran"